nemo/export/quantize/quantizer.py

```diff
@@ -167,22 +98,14 @@ def dummy():
         set_data_parallel_group(mpu.get_data_parallel_group())
         set_tensor_parallel_group(mpu.get_tensor_model_parallel_group())
 
-    @staticmethod
-    def _load_and_modify_config(
-        model_file: str,
-        tensor_model_parallel_size: Optional[int] = None,
-        pipeline_model_parallel_size: Optional[int] = None,
-    ):
-        model_cfg = load_config(model_file)
+        model.set_inference_config(OmegaConf.to_container(inference_config))
 
+    @staticmethod
+    def modify_model_config(model_cfg: DictConfig) -> DictConfig:
         with open_dict(model_cfg):
             model_cfg.activations_checkpoint_method = None
             model_cfg.activations_checkpoint_granularity = None
             model_cfg.sequence_parallel = False
```

Comment:
@janekl do you remember why sequence parallel is set to False? I tested for QAT and didn't have any issues with sequence parallel

Reply:
There is a functional issue during evaluation with sequence parallel=True. I double checked now and the code errors out here https://github.com/NVIDIA/Megatron-LM/blob/c4d12e26b2dc25a2eab7da92e2ac30338c0ed3de/megatron/core/model_parallel_config.py#L281. So it seems that single-GPU tests won't work with sequence parallelism enabled.

Related examples where it is set to False explicitly on evaluation (see also other settings):

1. https://github.com/NVIDIA/NeMo/blob/b0f3138a6be7fab3175deb8935f8492aeb1445bd/examples/nlp/language_modeling/megatron_gpt_eval.py#L226

2. https://github.com/EleutherAI/lm-evaluation-harness/blob/1060b68d35f6a4f1b571e7b5241049ead64b2936/lm_eval/models/nemo_lm.py#L52

Reply:
Sequence parallel is a technique used only for training, not for evaluation or inference (It requires the sequence length to be divisble by the TP size). This wont be true during token by token inference.

Reply:
Do we care how it is handled in evaluation though? Model can be PTQ/QAT on sequence parallel or with activations checkpointing enabled since it reduces memory requirement. When users run inference/eval on these models, its fine for these scripts to explicitly disable it. 
Plus users always have the option to disable sequence parallel if they want to by setting `model.sequence_parallel=False` in their config files.

Reply:
sequence_parallel could break PTQ (if the calibration seq len is not divisble by TP size). Hence I recommend disabling it by default. 
activation_checkpointing only matters for QAT, but does not affect PTQ. It should be okay to enable it by default.

Reply:
Got it. Thanks for the clarification. I will keep sequence_parallel update here and remove activations checkpointing update so users can set it what they want to. I will keep them on by default in the QAT example yaml as well